src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -2995,4 +2996,56 @@ static private boolean setsidExists() {
         }
         return false;
     }
+
+    /** {@inheritDoc} */
+    @Override
+    public Set<GitObject> getTags() throws GitException, InterruptedException {
+        ArgumentListBuilder args = new ArgumentListBuilder("show-ref", "--tags", "-d");
+        String result;
+        try {
+            result = launchCommandIn(args, workspace);
+        } catch (GitException ge) {
+            /* If no tags, then git show-ref --tags -d returns non-zero */
+            result = "";
+        }
+
+        /*
+        Output shows SHA1 and tag with (optional) marker for annotated tags
+        7ac27f7a051e1017da9f7c45ade8f091dbe6f99d refs/tags/git-3.6.4
+        7b5856ef2b4d35530a06d6482d0f4e972769d89b refs/tags/git-3.6.4^{}
+         */
+        String[] output = result.split("[\\n\\r]");
+        if (output.length == 0 || (output.length == 1 && output[0].isEmpty())) {
+            return new HashSet<>();
+        }
+        Pattern pattern = Pattern.compile("(\\p{XDigit}{40})\\s+refs/tags/([^\\^]*)(\\^\\{\\})?");
+        Map<String, ObjectId> tagMap = new HashMap<>();
+        for (String line : output) {
+            Matcher matcher = pattern.matcher(line);
+            if (!matcher.find()) {
```

Comment:
Do you really want to use `find` instead of `matches`? According to man git-show-ref the format is ` <SHA-1 ID> <space> <reference name>` and if anything else appears in the line it is probably an error to accept it.

Reply:
I've already seen more cases of git output changing unexpectedly than I want.  I'm leaving it more lax intentionally.

Reply:
In general, if the output changes, you're likely using the wrong git command, and should be parsing one of the plumbing commands instead. Changes have happened in the past. However, being slightly lax here isn't really a problem.

Git show-ref is definitely plumbing and thus unlikely to change.